### PR TITLE
Js2

### PR DESCRIPTION
**Pull request by roxannefranco**
Done with required features and still working on minor things. It would be great to have your feedback so I can implement changes on time for delivery! Thank you!

### CODE REVIEW

File: js/api/deletePost.mjs [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { url } from "../config.mjs";
+import { getToken } from "../functions.mjs";
+
+/**
+ * Delete a post
+ * @param {string} id
+ * @returns
```
Comment (Fermain) [inline js/api/deletePost.mjs]:
```suggestion
```
You can omit this if there is no return value

File: js/api/deletePost.mjs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      method: "DELETE",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: getToken(),
+      },
+    });
+    const data = await response.json();
+    return data;
+  } catch (e) {
+    console.log(e);
```
Comment (Fermain) [inline js/api/deletePost.mjs]:
```suggestion
    console.warn(e);
    alert("There was a problem deleting this post.")
```

**Review by Fermain — APPROVED** [review]
This is a well styled application that meets the requirements. User testing was smooth and I did not run into issues using the app.

There is a little bit missing from the UX side - in particular a loading indicator. This would help situate the user while they are waiting for something to happen.

On the same note, error handling could be a bit more sophisticated even if this means using an `alert` when something goes wrong. It's important to let the user know that something failed so they can try again.

**Comment by roxannefranco** [comment]
Thanks so much for your feedback Oliver! I will take it into consideration and make the changes necessary!

File: js/pages/feed.mjs [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+  // when clicked user signs out
+  signOutBtn.onclick = signOut;
+
+  let allPosts = [];
+  let term = "";
+
+  /**
+   * Fetch posts
+   */
+  async function getPosts() {
```
Comment (Fermain) [inline js/pages/feed.mjs:29]:
```suggestion
  async function getPosts(posts) {
```

**Pull request merged**